ci: Update rust version to 1.72

ecpullen opened the pull request:
<!--
Tips:
- Please read CONTRIBUTING.md to understand our process and our requests for PRs.
- Please file an issue before creating a PR so we can discuss the change and confirm it's not already being worked on.
-->

**Issue number:**



**Description of changes:**



**Testing done:**



**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

ecpullen committed:
ci: Update rust version to 1.72

stmcginnis reviewed (approved):
Looks good. We still have an issue with the check jobs, but this is still something good to do to stay up to date.

stmcginnis committed:
Fix new clippy lints for 1.72

The newer version of the rust toolchains enables some additional lints
that were not flagged in the previous version we were using. This
updates those places to make clippy happy.

Signed-off-by: Sean McGinnis <stmcg@amazon.com>
